pkg/api/deserialize_test.go

```diff
@@ -7,6 +7,62 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func TestMarshalJsonIngressType(t *testing.T) {
+	tests := map[string]IngressType{
```

Comment:
It looks like all these tests are running a set of test cases. Would you be willing to restructure these so they use subtests? They offer some benefits, see:  https://go.dev/blog/subtests . Example in our codebase:  https://github.com/G-Research/armada/blob/5c78c6bec2802738852f434328cde183f199a2a3/internal/lookout/repository/utils_test.go#L262

Reply:
Sure thing!